app/kumactl/pkg/cmd/root_context.go

```diff
@@ -46,7 +52,34 @@ func DefaultRootContext() *RootContext {
 			NewDataplaneTokenClient:    tokens.NewDataplaneTokenClient,
 			NewAPIServerClient:         kumactl_resources.NewAPIServerClient,
 		},
+		TypeArgs: map[string]core_model.ResourceType{
```

Comment:
Is it possible to use the same registry pattern? I mean register type in some global map inside `init()` method?